spec.md

```diff
@@ -100,6 +110,18 @@ interface ExpressionStatement <: Statement {
 
 An expression statement, i.e., a statement consisting of a single expression.
 
+## Directive
+
+```js
+interface Directive <: ExpressionStatement {
+    directive: string;
+}
+```
+
+A directive from the directive prologue of a script or function.
+The `directive` property is the raw string source of the directive.
+This source includes single or double wrapping quotes and may not contain an EscapeSequence or LineContinuation..
```

Comment:
> many cases map semantically different pieces of programs to identical ASTs

Each such case would be a spec bug. However, I'm unaware of any not already captured as issues (e.g., #6 and #98), and those hardly count as "many cases". Could you provide some examples?

> Just standardise raw.

I'm in favor of an optional—it must be optional to maintain backwards compatibility—`Literal#raw`, but note that it does come with the very same data duplication problems being discussed here ("[AST transformations… will need to synchronize all the places that represent essentially the same value in different forms](https://github.com/estree/estree/pull/99#discussion_r36993683)"). _And_ we would **still** need to introduce a `directive` property (e.g., for differentiating `"use strict"` from the parenthesized and therefore directive-free `("use strict")`), but it could be boolean.

So if we go that route (adding boolean-bearing `Directive <: ExpressionStatement` and string `Literal#raw`), code to check for strict mode would look like
```js
function isStrict( statementList ) {
  for ( let i = 0, statement, directive; (statement = statementList[i]); i++ ) {

    // End of Directive Prologue?
    if ( !statement.directive ) { break; }

    // Get raw string, defaulting to backwards-compatible `value`
    // (assuming that `directive` ⇒ `raw != null || raw.slice(1, -1) === value`)
    directive = (statement.expression.raw ||
        ("'" + statement.expression.value + "'")).slice(1, -1);

    // Require exact match
    if ( directive === "use strict" ) { return true; }
  }

  // No "use strict" directive
  return false;
}
```

... which has a more complicated body than the alternative
```js
    if ( !statement.directive ) { break; }
    if ( statement.directive === "use strict" ) { return true; }
```
but moves us closer to resolving #41, and gracefully condenses EscapeSequence- and LineContinuation-free string Literals wherever they appear.